yarn-project/simulator/src/public/public_tx_simulator.ts

```diff
@@ -0,0 +1,263 @@
+import {
+  type AvmProvingRequest,
+  type GasUsed,
+  type MerkleTreeReadOperations,
+  type NestedProcessReturnValues,
+  type SimulationError,
+  type Tx,
+  TxExecutionPhase,
+} from '@aztec/circuit-types';
+import { type GlobalVariables, type Header, type RevertCode } from '@aztec/circuits.js';
+import { type DebugLogger, createDebugLogger } from '@aztec/foundation/log';
+import { Timer } from '@aztec/foundation/timer';
+
+import { EnqueuedCallSimulator } from './enqueued_call_simulator.js';
+import { type PublicExecutor } from './executor.js';
+import { type WorldStateDB } from './public_db_sources.js';
+import { type PublicKernelCircuitSimulator } from './public_kernel_circuit_simulator.js';
+import { PublicKernelTailSimulator } from './public_kernel_tail_simulator.js';
+import { PublicTxContext } from './public_tx_context.js';
+import { generateAvmCircuitPublicInputs, runMergeKernelCircuit } from './utils.js';
+
+export type ProcessedPhase = {
+  phase: TxExecutionPhase;
+  durationMs: number;
+  returnValues: NestedProcessReturnValues[];
+  reverted: boolean;
+  revertReason?: SimulationError;
+};
+
+export type PublicTxResult = {
+  avmProvingRequest: AvmProvingRequest;
+  /** Gas used during the execution of this tx */
+  gasUsed: GasUsed;
+  revertCode: RevertCode;
+  /** Revert reason, if any */
+  revertReason?: SimulationError;
+  processedPhases: ProcessedPhase[];
+};
+
+export class PublicTxSimulator {
+  private log: DebugLogger;
+
+  constructor(
+    private db: MerkleTreeReadOperations,
+    private publicKernelSimulator: PublicKernelCircuitSimulator,
+    private globalVariables: GlobalVariables,
+    private worldStateDB: WorldStateDB,
+    private enqueuedCallSimulator: EnqueuedCallSimulator,
+    private publicKernelTailSimulator: PublicKernelTailSimulator,
+  ) {
+    this.log = createDebugLogger(`aztec:sequencer`);
```

Comment:
maybe something else than 'aztec:sequencer'?

Reply:
changed in stacked pr